src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseAsyncInAsync.cs

```diff
@@ -0,0 +1,265 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    /// <summary>
+    /// This analyzer recognizes invocations of JoinableTaskFactory.Run(Func{Task}), JoinableTask.Join(), and variants
+    /// that occur within an async method, thus defeating a perfect opportunity to be asynchronous.
+    /// </summary>
+    /// <remarks>
+    /// <![CDATA[
+    ///   async Task MyMethod()
+    ///   {
+    ///     JoinableTaskFactory jtf;
+    ///     jtf.Run(async delegate {  /* This analyzer will report warning on this JoinableTaskFactory.Run invocation. */
+    ///       await Stuff();
+    ///     });
+    ///   }
+    /// ]]>
+    /// </remarks>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class UseAsyncInAsync : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA2018";
+        internal const string AsyncMethodKeyName = "AsyncMethodName";
+        internal const string ExtensionMethodNamespaceKeyName = "ExtensionMethodNamespace";
+        internal const string MandatoryAsyncSuffix = "Async";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.CallAsyncMethodInAsyncContextTitle), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.CallAsyncMethodInAsyncContextMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.CallAsyncMethodInAsyncContextMessage), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Descriptor = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableMessage,
+                                                                                      DiagnosticCategory.Reliability,
+                                                                                      RuleLevel.BuildWarning,
```

Comment:
We should decrease this to `Info`. Here's the reasoning:

- This has a very high likelihood of having "noise" in the runtime repo
- Warning needs to be treated as a breaking change, and RC1 is late for a warning on something with probability of "noise"
- There are likely a lot of places where the "noise" is the right thing to do, such as async methods that accept a `bool useAsync` and then there are 2 branches of code

I will talk with our ASP.NET partners to let them know we'll use `Info` instead of `Warning`.

Reply:
Same concern as Jeff